src/components/Calculator.js

```diff
@@ -1,76 +1,45 @@
 import React from 'react';
 import '../App.css';
-import oprate from './operate';
+import calc from './logic/calculate';
 
-export default class myComponent extends React.Component {
+export default class calculator extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      firstNumber: '',
-      secondNumber: '',
-      operation: '',
-      result: '0',
+      total: null,
+      next: null,
+      operation: null,
     };
     this.CalculatorHandler = this.CalculatorHandler.bind(this);
-    this.AcHandler = this.AcHandler.bind(this);
-    this.ResultHandler = this.ResultHandler.bind(this);
   }
 
-  CalculatorHandler(value) {
-    const {
-      firstNumber, operation,
-    } = this.state;
-
-    if ((value === '+' || value === '-' || value === '×' || value === '÷' || value === '%')
-              && firstNumber !== '') {
-      this.setState(() => ({ operation: value }), () => {
-        this.setState((prevState) => ({ result: prevState.firstNumber + prevState.operation }),
-          () => { });
-      });
-    } else if (operation === '') {
-      this.setState((prevState) => ({ firstNumber: prevState.firstNumber + value }), () => {
-        this.setState((prevState) => ({ result: prevState.firstNumber }), () => {
-        });
-      });
-    } else if (operation !== '' && firstNumber !== '') {
-      this.setState((prevState) => ({ secondNumber: prevState.secondNumber + value }), () => {
-        this.setState((prevState) => (
-          { result: prevState.firstNumber + prevState.operation + prevState.secondNumber }),
-        () => { });
-      });
-    }
-  }
-
-  AcHandler() {
-    this.setState(
-      {
-        firstNumber: '', secondNumber: '', operation: '', result: '0',
-      },
-    );
+  setTotal(totalr) {
+    this.setState({ total: totalr });
   }
 
-  ResultHandler() {
-    const res = oprate.readEqual(this.state);
-    this.setState(() => ({ result: res }), () => {
-    });
-
-    this.setState(
-      {
-        firstNumber: '', secondNumber: '', operation: '',
-      },
-    );
+  CalculatorHandler(buttonName) {
+    const result = calc(this.state, buttonName);
+    this.setState(() => ({
+      total: result.total,
+      next: result.next,
+      operation: result.operation,
+    }));
```

Comment:
[OPTIONAL] Class methods are written in camelCase, it’s a best practice to actually tell what the function is doing by giving the function name a verb as prefix.
```js

// bad
name(firstName, lastName) {
  return `${firstName} ${lastName}`;
}

// bad
Name(firstName, lastName) {
  return `${firstName} ${lastName}`;
}
 
// good
getName(firstName, lastName) {
  return `${firstName} ${lastName}`;
}
```